### PR TITLE
[maestro] manually adjusting versions for .NET 7

### PR DESCRIPTION
**Pull request by jonathanpeppers**
We tried triggering subscriptions, such as:

    darc trigger-subscriptions --source-repo https://github.com/xamarin/xamarin-android --target-repo https://github.com/dotnet/Microsoft.Maui.Graphics --target-branch net7.0 --verbose

But then no PR opens?

So then I checked the build at:

https://maestro-prod.westus2.cloudapp.azure.com/2237/https:%2F%2Fgithub.com%2Fxamarin%2Fxamarin-android/latest/graph

Using the build id:

    darc update-dependencies --id 136370 --verbose

We get the error:

    fail: Microsoft.DotNet.Darc.Operations.Operation[0]
    Non-continuable HTTP 401 error encountered while making request
    System.Net.Http.HttpRequestException: Response status code does not indicate success: 401 (Unauthorized).
    at System.Net.Http.HttpResponseMessage.EnsureSuccessStatusCode()
    at Microsoft.DotNet.DarcLib.HttpRequestManager.ExecuteAsync(Int32 retryCount) in /_/src/Microsoft.DotNet.Darc/src/DarcLib/Helpers/HttpRequestManager.cs:line 97
    fail: Microsoft.DotNet.Darc.Operations.Operation[0]
    Something failed while trying the fetch the dependencies of repo 'https://dev.azure.com/dnceng/internal/_git/dotnet-optimization'

Trial and error, I found removing this element in
`Version.Details.xml` solved it:

    <Dependency Name="Microsoft.NET.Workload.Emscripten.Manifest-6.0.300" Version="6.0.4" CoherentParentDependency="Microsoft.NETCore.App.Ref">

I switched to `Microsoft.NET.Workload.Emscripten.Manifest-7.0.100`
that xamarin-android is using here:

https://github.com/xamarin/xamarin-android/blob/6568f901a5b1a3829d8d5a5254eddd62a1765b95/eng/Version.Details.xml#L15-L18

And then things worked! I think Maestro updates will start working
after this goes in.

I also updated the `CoherentParentDependency` on:

    <Dependency Name="Microsoft.NETCore.App.Ref" Version="7.0.0-preview.5.22258.4" CoherentParentDependency="Microsoft.Dotnet.Sdk.Internal">

To match what xamarin-android has.

### CODE REVIEW

**Review by mattleibow — APPROVED** [review]

File: eng/pipelines/common/provision.yml [hunk 1]
```diff
@@ -83,7 +83,9 @@ steps:
         echo "##vso[task.setvariable variable=MSBUILD_EXE]$msbuild"
       displayName: 'Setup MSBuild Paths'
       condition: eq(variables['provisioningVS'], 'true')
-
+    - pwsh: |
+        'y' | & "${env:ProgramFiles(x86)}\Android\android-sdk\cmdline-tools\5.0\bin\sdkmanager" "platforms;android-32"
+      displayName: Install Android API 32
```
[the remark this answers is not in the document]
Comment (jonathanpeppers) [reply eng/pipelines/common/provision.yml:88]: I found the steps running provisionator were disabled. When I tried enabling them, it was missing credentials...

**Pull request merged**